samples/errors7.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+)
+
+func writeCheck() {
+
+	var err error
+
+	buf := make([]byte, 0, 0)
+	write := func(n byte) {
```

Comment:
a standalone function is easier to understand imo.

this sample has me confused. I'll wait for others to weigh in. @anzboi, @camh-anz ?

Reply:
I basically want to illustrate use of a closure to avoid multiple error checks. My first sample which was only a snippet of code kind of had the entire picture but the code didn't compile. When I wrote the whole code with Append and AppendClosure (which compiled), it kind of became big and difficult to explain. Then I resorted to only this now it's a bit back to square one (bit hard to comprehend). I'll see if I can think of something. And wait for others to weigh in.

Reply:
Does everyone think that it is worth mentioning a solution via a pattern to avoid repetitive error checks ? I have now these two which I can show side by side. This code is better than the ones I had before.

https://goplay.space/#bTUOIQOSR1t
https://goplay.space/#IBBy8S2-GBx

Reply:
The closure over the `err` var form for error handling is something I've used too, and I'm happy with what's here. I think the example storing `err` in a struct is over-done. I learnt it from that same blog article.

When presenting, it would be good to call out exactly what is going on here as it will likely cause some confusion with others. Perhaps add some presenter notes for this?

Reply:
Thank you. I have pushed errors7a.go and errors7b.go which I will plan to show side-by-side and illustrate only the important parts.